plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java

```diff
@@ -201,6 +201,11 @@ public void uploadMLModel(MLUploadInput uploadInput, MLTask mlTask) {
                         log.info("create new model meta doc {} for upload task {}", modelId, taskId);
                         modelHelper.downloadAndSplit(modelId, modelName, version, uploadInput.getUrl(), ActionListener.wrap(result -> {
                             Long modelSizeInBytes = (Long) result.get(MODEL_SIZE_IN_BYTES);
+                            if (modelSizeInBytes/1024/1024/1024 > 4L) {
```

Comment:
Will 4.6 GB be truncated as 4GB in this way?

Reply:
That depends on how accurate we want to limit it. I think >= 4L will resolve your concern. Anything equals to 4 after truncation will be treated as oversize.

Reply:
If a file size is 4.5 GB, this line will truncate it to 4GB,  it won't throw exception because the truncated file size is not greater than the limit `4GB` ?

Reply:
I already change to >= 4GB. So 4.5GB will be truncated to 4, which makes >= 4GB is true and throw an exception.